File: Problems/C/014.py
from itertools import accumulate
import sys
input = lambda: sys.stdin.readline().rstrip()
num = 1000005
n = int(input())
imos = [0 for _ in range(num)]

for i in range(n):
    a, b = map(int, input().split())
    imos[a] += 1
    imos[b + 1] -= 1

# for i in range(1, num):
#     imos[i] += imos[i - 1]
ans = accumulate(imos)

print(max(ans))

File: Problems/C/014.cpp
#define _GLIBCXX_DEBUG
#include <bits/stdc++.h>
using namespace std;
typedef long long ll;
typedef pair<int, int> P;
#define rep(i, n) for(int i = 0; i < (int)(n); i++)
#define all(v) (v).begin(), (v).end()

int main() {
  int n;
  cin >> n;
  int num = 1000005;
  vector<int> imos(num);
  rep(i, n) {
    int a, b;
    cin >> a >> b;
    imos[a]++;
    imos[b + 1]--;
  }
  int ans = 0;
  rep(i, num) {
    if(i > 0) imos[i] += imos[i - 1];
    ans = max(ans, imos[i]);
  }
  cout << ans << endl;
  return 0;
}
